docs/paper/dev/api/event-api/chat-event.md

```diff
@@ -0,0 +1,138 @@
+- The `message` parameter is the message that was sent.
+- The `viewer` parameter is the player that is receiving the message.
+
+:::tip
+
+#### ChatRenderer.ViewerUnaware
+
+If your renderer does not need to know about the viewer, you can use the `ChatRenderer.ViewerUnaware` interface instead of the `ChatRenderer` interface. 
+This will benefit performance as the message will only be rendered once instead of once for each viewer.
+
+:::
+
+Now that we understand how the renderer works, we can start using it.
+
+## Using the renderer
+
+There are two ways to use the renderer.
+1. Implementing the `ChatRenderer` interface in a class.
+2. Using a lambda expression.
+
+Depending on the complexity of your renderer you may want to use one or the other.
+
+### Implementing the ChatRenderer interface
+
+The first way of using the renderer is by implementing the `ChatRenderer` interface in a class.
+In this example, we will be using our ChatListener class. 
+Next, we need to tell the event to use the renderer by using the `renderer` method.
+
+```java title="ChatListener.java"
+public class ChatListener implements Listener, ChatRenderer { // Implement the ChatRenderer and Listener interface
+
+    // Listen for the AsyncChatEvent
+    @EventHandler
+    public void onChat(AsyncChatEvent event) { 
+        event.renderer(this); // Tell the event to use our renderer
+    }
+    
+    // Override the render method
+    @Override
+    public @NotNull Component render(@NotNull Player source, @NotNull Component sourceDisplayName, @NotNull Component message, @NotNull Audience viewer) {
+        // ...
+    }
+    
+}
+```
+
+:::note
+
+If you decide to create a separate class for your renderer, it is important to know that you don't need to instantiate the class every time the event is called.
+In this case you can use the singleton pattern to create a single instance of the class.
+
+:::
+
+### Using a lambda expression
+
+Another way of using the renderer is by using a lambda expression.
+
+```java title="ChatListener.java"
+public class ChatListener implements Listener {
+
+    @EventHandler
+    public void onChat(AsyncChatEvent event) {
+        event.renderer((source, sourceDisplayName, message, viewer) -> {
+            // ...
+        });
+    }
+
+}
+```
+
+## Rendering the message
+
+Now that we have our renderer, we can start rendering the message.
+
+Let's say we want to render our chat to look like this:
+
+![](./assets/plain-message-rendering.png)
+
+To do this, we need to return a new `Component` that contains the message we want to send.
+
+```java title="ChatListener.java"
+public class ChatListener implements Listener, ChatRenderer {
+
+    @Override
+    public @NotNull Component render(@NotNull Player source, @NotNull Component sourceDisplayName, @NotNull Component message, @NotNull Audience viewer) {
+        return sourceDisplayName 
+                .append(Component.text(": ")) 
+                .append(message); 
+    }
+
+}
+```
```

Comment:
Missing the listener here?

Reply:
Intended since it didn't change and would just distract from the changes in `render`